### ISSUE TITLE
(🐞) 'warn-unreachable' doesn't reactivate after it is suppressed with a `<nothing>` statement

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11437]
```py
from typing import NoReturn

def foo() -> NoReturn:
    raise Exception

def bar() -> NoReturn:
    foo()
    foo()
    print("hi")  # no warning
```

The first call to `foo` is suppress the `unreachable` intentionally, (same with the second). But the `print("hi)` should definitely show an unreachable error.

**Comment by sobolevn** [comment]
Looks like this is intentional: https://github.com/python/mypy/blame/5703becbbe9be5006ed0d2d998dd4a793f1aedd0/test-data/unit/check-unreachable-code.test#L905

It can be easily changed, but I am not sure that this won't break a lot of existing code.
Let's try it out.

**Comment by erictraut** [comment]
It appears that the latest version of mypy (1.5) behaves as expected here, so I think this is safe to close.

**Comment by ikonst** [comment]
Likely fixed by #15386
